pnpm/src/main.ts

```diff
@@ -200,7 +200,7 @@ export async function main (inputArgv: string[]): Promise<void> {
     const filterResults = await filterPackagesFromDir(wsDir, filters, {
       engineStrict: config.engineStrict,
       nodeVersion: config.nodeVersion ?? config.useNodeVersion,
-      patterns: cliOptions['workspace-packages'],
+      patterns: cliOptions['workspace-packages'] ?? config.workspaceManifest?.contents.packages,
```

Comment:
Add a new setting to the config object that will prioritize the `workspace-packages` cli flag over the. packages from workspace manifest. Here you'll just need to do 
```suggestion
      patterns: config.workspacePackagePatterns,
```

Reply:
I can add the new `config.workspacePackagePatterns` field, but do you mind if I fold the `cliOptions['workspace-packages']` in a separate PR?

I just want to do a bit more testing to ensure there's no behavior changes. Today, only the `--recursive` flag here reads `cliOptions['workspace-packages']`. I want to make sure things don't unexpectedly break if that CLI option takes priority _always_.

Reply:
It is fine if it will take priority always.

Reply:
Can I do this in another PR? I wanted this PR to just be a refactor without behavior changes. The change above might subtly change how pnpm behaves.

Reply:
I understand we want `cliOptions['workspace-packages']` to take precedence, but the problem is it doesn't today in all cases. Fixing that might introduce other bugs.

Reply:
It doesn't matter. That flag was added for one specific user (who seems like moved to bun) and we don't even have the flag documented on the website.

You can do a separate PR but nobody will notice the difference.

Reply:
Gotcha. I was probably a bit too afraid of risk in this case. I'm glad to hear nobody will notice either way.

Reply:
Making this change in https://github.com/pnpm/pnpm/pull/8215.